GetIntoTeachingApi/Models/TeachingEventSearchRequest.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using GetIntoTeachingApi.Services;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace GetIntoTeachingApi.Models
+{
+    public class TeachingEventSearchRequest
+    {
+        [SwaggerSchema("Postcode to center search around.")]
+        public string Postcode { get; set; }
+        [SwaggerSchema("Set to filter results to a radius (in miles) around the postcode.")]
+        public int? Radius { get; set; }
+        [SwaggerSchema("Set to filter results to a type of teaching event. Must match an `typeId` of the `TeachingEvent` schema.")]
+        public int? TypeId { get; set; }
+        [SwaggerSchema("Set to filter results to those that start after a given date.")]
+        public DateTime? StartAfter { get; set; }
+        [SwaggerSchema("Set to filter results to those that start before a given date.")]
+        public DateTime? StartBefore { get; set; }
+
+        public bool Match(TeachingEvent teachingEvent, IPostcodeService postcodeService)
+        {
+            if (TypeId != null && TypeId != teachingEvent.TypeId)

```

Comment:
Omitting the { brackets is generally seen as bad practice.  Im sure resharper disagrees with me

Reply:
Hah yeah I'm afraid it does - I agree usually but since C# wants the brackets on a separate line it irks me 😄